app/views/uploads/index.html.erb

```diff
@@ -1,31 +1,37 @@
-<%= render 'shared/organization_header' %>
+<%= render 'shared/layout_upload_page' do %>
 
-<div class="container">
-  <h1>Uploads</h1>
+  <div class="d-flex align-items-center">
+
+    <h2 class="font-weight-normal mb-0">Uploads</h2>
+
+    <div class="ms-auto p-2">
+      <span class="ms-4"><%= link_to 'New upload', new_organization_upload_path(@stream.organization, stream: (@stream.default? ? nil : @stream.friendly_id)), class: 'btn btn-primary' if can? :create, @stream.uploads.build %></span>
+    </div>
+
+  </div>
+
+  <table class="table table-striped mt-5">
 
-  <table class="mt-4">
     <thead>
       <tr>
         <th>Name</th>
-        <th>Files</th>
-        <th colspan="3"></th>
+        <th>Date</th>
+        <th>Stream</th>
+        <th>Actions</th>
       </tr>
     </thead>
 
     <tbody>
-      <% @uploads.each do |upload| %>
+      <% @uploads.sort_by(&:created_at).reverse.each do |upload| %>
         <tr>
-          <td><%= upload.name %></td>
-          <td><%= upload.files %></td>
-          <td><%= link_to 'Show', [@organization, upload] %></td>
-          <td><%= link_to 'Edit', edit_organization_upload_path(@organization, upload) %></td>
-          <td><%= link_to 'Destroy', [@organization, upload], method: :delete, data: { confirm: 'Are you sure?' } %></td>
+          <td><%= link_to upload.name, [@organization, upload] %></td>
+          <td><%= local_time(upload.created_at, format: datetime_display_format()) %></td>
+          <td><%= link_to @stream.display_name, organization_stream_path(@stream.organization, @stream) %></td>
```

Comment:
I don't think this is quite right. We need the stream associated with the upload -- maybe something like:

```suggestion
          <td><%= link_to upload.stream.display_name, organization_stream_path(upload.stream.organization, upload.stream) %></td>
```

Reply:
Ahh this was leftover from the old code! Thanks.

Reply:
@corylown actually I take this back - I added the @stream variable in the controller. It should be accurate as is. ?

Reply:
I don't think the controller variable works in this context because the controller only grabs one stream (haven't looked closely at the logic, but in the table you're showing uploads from all different streams so you need to show the stream associated with each separate upload.

Reply:
Ok makes sense thank you.

Reply:
@corylown actually you are right below; fixing this stream issue now breaks the New upload button. I am going to remove it from the layout.